tstest/integration/integration_test.go

```diff
@@ -258,10 +258,21 @@ func TestAddPingRequest(t *testing.T) {
 	}
 
 	// Wait for PingRequest to come back
-	pingTimeout := time.NewTimer(10 * time.Second)
+	waitDuration := 10 * time.Second
+	pingTimeout := time.NewTimer(waitDuration)
+
+	// Ticker sends new PingRequests if the previous did not get through
+	ticker := time.NewTicker(waitDuration / 5)
+
 	select {
 	case <-gotPing:
 		pingTimeout.Stop()
+		ticker.Stop()
+	case <-ticker.C:
+		ok := env.Control.AddPingRequest(nodeKey, pr)
```

Comment:
This "fix" feels like a hacky workaround rather than understanding the problem.

Why is a retry even necessary? The earlier `AddPingRequest` must've worked if it didn't fail at line 257, so where did it go?